src/blocks/file-blocks/geojson.tsx

```diff
@@ -0,0 +1,319 @@
+      if (!geojsonObject.features) {
+        geojsonObject = {
+          type: "FeatureCollection",
+          features: [geojsonObject],
+        };
+      }
+      geojsonObject = {
+        ...geojsonObject,
+        features: geojsonObject.features.map((d, i) => ({
+          ...d,
+          properties: { ...d.properties, __geojson_internal_index__: i },
+        })),
+      };
+      return geojsonObject;
+    } catch (e) {
+      return {};
+    }
+  }, [modifiedContent]);
+
+  const features = useMemo(() => parseGeoJSON(geojson), [geojson]);
+
+  const hoveredFeature = features.find(
+    (f) => f.values_.__geojson_internal_index__ === hoveredFeatureId
+  );
+
+  return (
+    <div className="relative w-full h-full overflow-auto">
+      {isMounted && (
+        <RMap
+          className="w-full h-full min-h-[10em] min-w-[10em]"
+          initial={{
+            center: [0, 0],
+            zoom: 1,
+          }}
+          onClick={(event) => {
+            setIsHoveredFeatureLocked(false);
+            setHoveredFeatureId(null);
+          }}
+        >
+          {/* tileset */}
+          <ROSM />
+
+          {/* geojson shapes */}
+          <RLayerVector
+            zIndex={10}
+            features={features}
+            onPostRender={(event) => {
+              // only do this once
+              try {
+                const extent = event.target.values_.source.getExtent();
+                if (lastExtent.join(",") === extent.join(",")) return;
+                setLastExtent(extent);
+                // zoom out a bit
+                const zoomedOutExtent = [
+                  extent[0] - (extent[2] - extent[0]) * 0.1,
+                  extent[1] - (extent[3] - extent[1]) * 0.1,
+                  extent[2] + (extent[2] - extent[0]) * 0.1,
+                  extent[3] + (extent[3] - extent[1]) * 0.1,
+                ];
+                const map = event.target.values_.map;
+                const view = map?.getView();
+                if (!view) return;
+                view.fit(zoomedOutExtent, {
+                  duration: 350,
+                  maxZoom: 15,
+                });
+              } catch (e) {
+                console.log(e);
+              }
+            }}
+            onPointerMove={(e) => {
+              if (isHoveredFeatureLocked) return;
+              if (
+                hoveredFeatureId === e.target.values_.__geojson_internal_index__
+              )
+                return;
+              setHoveredFeatureId(e.target.values_.__geojson_internal_index__);
+            }}
+            onPointerLeave={(e) => {
+              if (isHoveredFeatureLocked) return;
+              hoveredFeatureId ===
+                e.target.values_.__geojson_internal_index__ &&
+                setHoveredFeatureId(null);
+            }}
+            onClick={(event) => {
+              event.stopPropagation();
+              setIsHoveredFeatureLocked((d) => !d);
+            }}
```

Comment:
little nit — if you hover a region, click to lock, then hover a different region and click / unclick it unlocks / locks the first region. Maybe `onClick` should unset the lock if you're in the already-locked region, otherwise lock the current region.

Reply:
absolutely agree! I didn't want to over-complicate it (this'll require another piece of state), but yeah I think the better ux is worth it! will add

Reply:
no, that's a lie! I was thinking of a different UX. no new state needed 🎉

Reply:
updated! feels great